r2dec compilation failed

**Describe the bug**

Tried to install r2dec on latest Kali weekly VM:

```
git clone https://github.com/radareorg/radare2.git external/radare2
cd external/adare2
sys/install.sh
r2pm -U
sudo apt-get install meson
r2pm -ci r2dec
```
The last step is failing with the following output:

```
┌──(kali㉿kali)-[~/emba/external/radare2]
└─$ r2pm -ci r2dec            
INFO: Cleaning r2dec
INFO: Using r2-5.9.9 and r2pm-5.9.9
INFO: Cleaning r2dec
INFO: git clone --depth=1 --recursive https://github.com/wargio/r2dec-js /home/kali/.local/share/radare2/r2pm/git/r2dec
Cloning into '/home/kali/.local/share/radare2/r2pm/git/r2dec'...
remote: Enumerating objects: 95, done.
remote: Counting objects: 100% (95/95), done.
remote: Compressing objects: 100% (91/91), done.
remote: Total 95 (delta 2), reused 43 (delta 1), pack-reused 0 (from 0)
Receiving objects: 100% (95/95), 140.93 KiB | 3.35 MiB/s, done.
Resolving deltas: 100% (2/2), done.
INFO: Starting install for r2dec
INFO: R2PM_NEEDS: Found gcc in PATH
INFO: R2PM_NEEDS: Found make in PATH
INFO: R2PM_NEEDS: Found ninja in PATH
INFO: R2PM_NEEDS: Found meson in PATH
INFO: SCRIPT=<<EOF
INFO:   meson setup -Dr2_plugdir=${R2PM_PLUGDIR} b --backend=ninja --wipe || exit 1
        ninja -C b || exit 1
        ninja -C b install || exit 1
INFO: EOF
The Meson build system
Version: 1.7.0
Source dir: /home/kali/.local/share/radare2/r2pm/git/r2dec
Build dir: /home/kali/.local/share/radare2/r2pm/git/r2dec/b
Build type: native build
Project name: r2dec
Project version: undefined
C compiler for the host machine: cc (gcc 14.2.0 "cc (Debian 14.2.0-19) 14.2.0")
C linker for the host machine: cc ld.bfd 2.44
Host machine cpu family: x86_64
Host machine cpu: x86_64
Cloning into 'libquickjs'...
remote: Enumerating objects: 133, done.
remote: Counting objects: 100% (133/133), done.
remote: Compressing objects: 100% (118/118), done.
remote: Total 133 (delta 5), reused 59 (delta 4), pack-reused 0 (from 0)
Receiving objects: 100% (133/133), 816.31 KiB | 2.27 MiB/s, done.
Resolving deltas: 100% (5/5), done.

Executing subproject libquickjs 

libquickjs| Project name: quickjs-ng
libquickjs| Project version: v0.8.0
libquickjs| C compiler for the host machine: cc (gcc 14.2.0 "cc (Debian 14.2.0-19) 14.2.0")
libquickjs| C linker for the host machine: cc ld.bfd 2.44
libquickjs| Run-time dependency threads found: YES
libquickjs| Library atomic found: YES
libquickjs| Library dl found: YES
libquickjs| Library m found: YES
libquickjs| Compiler for C supports arguments -fvisibility=hidden: YES
libquickjs| Compiler for C supports arguments -fvisibility-inlines-hidden: NO
libquickjs| Compiler for C supports arguments -Wno-implicit-fallthrough: YES
libquickjs| Compiler for C supports arguments -Wno-sign-compare: YES
libquickjs| Compiler for C supports arguments -Wno-missing-field-initializers: YES
libquickjs| Compiler for C supports arguments -Wno-unused-parameter: YES
libquickjs| Compiler for C supports arguments -Wno-unused-but-set-variable: YES
libquickjs| Compiler for C supports arguments -Wno-array-bounds: YES
libquickjs| Compiler for C supports arguments -Wno-format-truncation: YES
libquickjs| Compiler for C supports arguments -Wno-format-zero-length: YES
libquickjs| Compiler for C supports arguments -funsigned-char: YES
libquickjs| Build targets in project: 1
libquickjs| Subproject libquickjs finished.

Program radare2 found: YES (/usr/local/bin/radare2)
Library r_core found: YES
Library r_util found: YES
Library r_cons found: YES
Library r_config found: YES
Library r_io found: YES
Library r_lang found: YES
Compiler for C supports arguments -fvisibility=hidden: YES (cached)
Compiler for C supports arguments -fvisibility-inlines-hidden: NO (cached)
Compiler for C supports link arguments -Wl,--exclude-libs,ALL: YES 
Message: radare2 include dir: /usr/local/include/libr
Message: radare2 library dir: /usr/local/lib
Message: radare2 plugin dir:  /home/kali/.local/share/radare2/plugins
Build targets in project: 6

r2dec undefined

  Subprojects
    libquickjs: YES

  User defined options
    backend   : ninja
    r2_plugdir: /home/kali/.local/share/radare2/plugins

Found ninja-1.12.1 at /usr/bin/ninja
ninja: Entering directory `b'                                                                                       
[14/16] Compiling C object libcore_pdd.so.p/c_r2dec-plugin.c.o
FAILED: libcore_pdd.so.p/c_r2dec-plugin.c.o 
cc -Ilibcore_pdd.so.p -I. -I.. -I../c -I/usr/local/include/libr -Isubprojects/libquickjs -I../subprojects/libquickjs -fvisibility=hidden -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -O0 -g -fPIC -fvisibility=hidden -flto -fvisibility=hidden -MD -MQ libcore_pdd.so.p/c_r2dec-plugin.c.o -MF libcore_pdd.so.p/c_r2dec-plugin.c.o.d -o libcore_pdd.so.p/c_r2dec-plugin.c.o -c ../c/r2dec-plugin.c
../c/r2dec-plugin.c:290:17: error: initialization of ‘_Bool (*)(RCorePluginSession *, const char *)’ {aka ‘_Bool (*)(struct r_core_plugin_session_t *, const char *)’} from incompatible pointer type ‘int (*)(void *, const char *)’ [-Wincompatible-pointer-types]
  290 |         .call = r_cmd_pdd,
      |                 ^~~~~~~~~
../c/r2dec-plugin.c:290:17: note: (near initialization for ‘core_plugin_r2dec.call’)
../c/r2dec-plugin.c:291:17: error: initialization of ‘_Bool (*)(RCorePluginSession *)’ {aka ‘_Bool (*)(struct r_core_plugin_session_t *)’} from incompatible pointer type ‘int (*)(void *, const char *)’ [-Wincompatible-pointer-types]
  291 |         .init = r_cmd_pdd_init
      |                 ^~~~~~~~~~~~~~
../c/r2dec-plugin.c:291:17: note: (near initialization for ‘core_plugin_r2dec.init’)
[15/16] Compiling C object libcore_pdd.so.p/c_r2dec.c.o
ninja: build stopped: subcommand failed.
```

--- a/c/r2dec-plugin.c
+++ b/c/r2dec-plugin.c
@@ -8,8 +8,6 @@
 
 #include "r2dec.h"
 
-typedef struct r_core_t RCore;
-
 typedef struct exec_context_t {
 	RCore *core;
 	void *bed;
@@ -231,21 +229,57 @@ static void _cmd_pdd(RCore *core, const char *input) {
 	}
 }
 
-static int r_cmd_pdd(void *user, const char *input) {
-	RCore *core = (RCore *) user;
-	if (r_str_startswith (input, "pdd")) {
 #if R2_VERSION_NUMBER >= 50909
+static bool r_cmd_pdd(RCorePluginSession *cps, const char *input) {
+	if (r_str_startswith (input, "pdd")) {
+		RCore *core = cps->core;
 		const ut64 addr = core->addr;
-#else
-		const ut64 addr = core->offset;
-#endif
 		_cmd_pdd (core, input + 3);
 		r_core_seek (core, addr, true);
 		return true;
 	}
 	return false;
 }
+#else
+static int r_cmd_pdd(void *user, const char *input) {
+	RCore *core = (RCore *) user;
+	if (r_str_startswith (input, "pdd")) {
+		const ut64 addr = core->offset;
+		_cmd_pdd (core, input + 3);
+		r_core_seek (core, addr, true);
+		return 1;
+	}
+	return 0;
+}
+#endif
 
+#if R2_VERSION_NUMBER >= 50909
+static bool r_cmd_pdd_init(RCorePluginSession *cps) {
+	RConfig *cfg = cps->core->config;
+	r_config_lock (cfg, false);
+	SETPREF("r2dec.asm", "false", "if true, shows pseudo next to the assembly.");
+	SETPREF("r2dec.blocks", "false", "if true, shows only scopes blocks.");
+	SETPREF("r2dec.casts", "false", "if false, hides all casts in the pseudo code.");
+	SETPREF("r2dec.debug", "false", "do not catch exceptions in r2dec.");
+	SETPREF("r2dec.highlight", "default", "highlights the current address.");
+	SETPREF("r2dec.paddr", "false", "if true, all xrefs uses physical addresses compare.");
+	SETPREF("r2dec.slow", "false", "load all the data before to avoid multirequests to r2.");
+	SETPREF("r2dec.xrefs", "false", "if true, shows all xrefs in the pseudo code.");
+	r_config_lock (cfg, true);
+
+	const char *commands[] = {
+		"pdd", "pdd?", "pdd*", "pdda", "pddb",
+		"pddc", "pddf", "pddi", "pdds", "pddu",
+		NULL
+	};
+	RCoreAutocomplete *a = cps->core->autocomplete;
+	for (const char **cmd = commands; *cmd; cmd++) {
+		r_core_autocomplete_add(a, *cmd, R_CORE_AUTOCMPLT_DFLT, true);
+		cmd++;
+	}
+	return true;
+}
+#else
 static int r_cmd_pdd_init(void *user, const char *cmd) {
 	RCmd *rcmd = (RCmd*) user;
 	RCore *core = (RCore *) rcmd->data;
@@ -274,6 +308,7 @@ static int r_cmd_pdd_init(void *user, const char *cmd) {
 	r_core_autocomplete_add (core->autocomplete, "pddu", R_CORE_AUTOCMPLT_DFLT, true);
 	return true;
 }
+#endif
 
 RCorePlugin core_plugin_r2dec = {
 #if R2_VERSION_NUMBER > 50808

